apis/config/v1alpha1/clientconfig_types.go

```diff
@@ -266,6 +266,14 @@ type GCPPluginRepository struct {
 	RootPath string `json:"rootPath,omitempty" yaml:"rootPath"`
 }
 
+// S3PluginRepository is a CLI plugin aws s3 block storage.
+type S3PluginRepository struct {
```

Comment:
We already have generic `GenericRESTDiscovery`. Why do we want to add `S3PluginRepository`?
Based on my understanding we should be able to use the same discovery to download artifacts from S3.

Also, `PluginRepository` and `GCPPluginRepository` is old config used before context-scoped plugin discovery. We will be cleaning some of those stuff as part of #3224

Reply:
Okay i guess i'll reuse the `genericRestDiscovery`. Sure we dont have any dependency on the `GCPPluginRepository` so should be good. Thanks.

Reply:
Good catch! I see that `S3PluginRepository` is used only to allow the S3 url in downloads. We could just hard code the S3 path in some constant & use it.